### ISSUE TITLE
Items in playlists cannot be moved. The item goes back in the interface back to its previous location and an error "Index was outside the bounds of the array" appears in the logs.

### ISSUE THREAD

**Issue by charliec111** [jellyfin/jellyfin#12892]
### This issue respects the following points:

- [X] This is a **bug**, not a question or a configuration issue; Please visit our forum or chat rooms first to troubleshoot with volunteers, before creating a report. The links can be found [here](https://jellyfin.org/contact/).
- [X] This issue is **not** already reported on [GitHub](https://github.com/jellyfin/jellyfin/issues?q=is%3Aopen+is%3Aissue) _(I've searched it)_.
- [X] I'm using an up to date version of Jellyfin Server stable, unstable or master; We generally do not support previous older versions. If possible, please update to the latest version before opening an issue.
- [X] I agree to follow Jellyfin's [Code of Conduct](https://jellyfin.org/docs/general/community-standards.html#code-of-conduct).
- [X] This report addresses only a single issue; If you encounter multiple issues, kindly create separate reports for each one.

### Description of the bug

After upgrading to 10.10.0, playlist items can not be moved. (Downgrading back to 10.9.11 solves the issues, so it seems likely due to the upgrade). I have previously had a similar issue where a playlist containing a removed item could no longer be rearranged or edited (that seemed like it was fixed). I tried creating new playlists, restarting the docker container, restarting the docker service and restarting the system. I also tried connecting through the local ip and http://  rather than through a nginx-proxy-manager and https://. Nothing solved the issue until I reverted the docker container to the 10.9.11 image.

Since part of the error in the logs said "Index was outside the bounds of the array," I tried moving an item to different places but all locations in a playlist led to the same error.

### Reproduction steps

1. Create a playlist and add items.
2. Rearrange the items in the jellyfin interface.

### What is the current _bug_ behavior?

The playlist order does not change and an error appears in the logs saying " [ERR] [31] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL POST /Playlists/c58b999842431004eb431fecfc2037f2/Items/be6a6882828d47dfba59eaa928003ca1/Move/0.
System.IndexOutOfRangeException: Index was outside the bounds of the array."

### What is the expected _correct_ behavior?

The playlist order changes.

### Jellyfin Server version

10.10.0+

### Specify the build version

10.10.0

### Environment

```markdown
- OS: Debian Bullseye
- Linux Kernel: 6.1.0-25
- Virtualization: Docker
- Clients: Browser, android, iOS
- Browser: Chrome
- FFmpeg Version:
- Playback Method: Direct Play
- Hardware Acceleration: none
- GPU Model:
- Plugins:
- Reverse Proxy: nginx-proxy-manager
- Base URL:
- Networking: 
- Storage: local
```

### Jellyfin logs

```shell
[17:31:46] [ERR] [31] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL POST /Playlists/c58b999842431004eb431fecfc2037f2/Items/be6a6882828d47dfba59eaa928003ca1/Move/0.
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Emby.Server.Implementations.Playlists.PlaylistManager.MoveItemAsync(String playlistId, String entryId, Int32 newIndex)
   at Jellyfin.Api.Controllers.PlaylistsController.MoveItem(String playlistId, String itemId, Int32 newIndex)
   at lambda_method1267(Closure, Object)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Jellyfin.Api.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Api.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Api.Middleware.IPBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Api.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Api.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Jellyfin.Api.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)
   at Jellyfin.Api.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```

**Comment by amanaedk** [comment]
I can reproduce this in the 10.10.0 and also cannot delete playlist items.

**Comment by DickieKnee** [comment]
Also can not remove items in playlist 
[https://i.postimg.cc/cJ3QZxH6/JFVer-Sion.png](url)
[https://i.postimg.cc/4dTpxD7P/Can-Not-Remove.png](url)

**Comment by RealHypnoticOcelot** [comment]
I'm having the same issue, but my logs don't throw any errors when I try to remove something from a playlist.

**Comment by pixel24** [comment]
Yes, I’ve been experiencing this bug since 10.10.0 as well, and it's been tough, especially because I spend a lot of time listening to music. I could technically roll back to version 10.9.11 on my PVE, but unfortunately, I only noticed the bug after adding a lot of new tracks to playlists. Now, I’d have to manually back up these playlists and merge them again after the rollback.

I'm really hoping this issue will be resolved soon.

**Comment by erictroebs** [comment]
Removing an item from a playlist does not raise an exception for me. Moving an item does.

<details>
<summary>error log using 10.10 docker image</summary>
```
jellyfin-1  | [14:15:11] [ERR] [52] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL POST /Playlists/169689a75cacce51f6e56688ec25c771/Items/1017fab661eb4f2e8295d545bf98cf59/Move/30.
jellyfin-1  | System.IndexOutOfRangeException: Index was outside the bounds of the array.
jellyfin-1  |    at Emby.Server.Implementations.Playlists.PlaylistManager.MoveItemAsync(String playlistId, String entryId, Int32 newIndex)
jellyfin-1  |    at Jellyfin.Api.Controllers.PlaylistsController.MoveItem(String playlistId, String itemId, Int32 newIndex)
jellyfin-1  |    at lambda_method1567(Closure, Object)
jellyfin-1  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
jellyfin-1  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
jellyfin-1  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
jellyfin-1  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
jellyfin-1  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
jellyfin-1  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
jellyfin-1  | --- End of stack trace from previous location ---
jellyfin-1  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
jellyfin-1  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
jellyfin-1  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
jellyfin-1  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
jellyfin-1  | --- End of stack trace from previous location ---
jellyfin-1  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
jellyfin-1  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
jellyfin-1  |    at Jellyfin.Api.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
jellyfin-1  |    at Jellyfin.Api.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
jellyfin-1  |    at Jellyfin.Api.Middleware.IPBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
jellyfin-1  |    at Jellyfin.Api.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
jellyfin-1  |    at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
jellyfin-1  |    at Jellyfin.Api.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)
jellyfin-1  |    at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
jellyfin-1  |    at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
jellyfin-1  |    at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
jellyfin-1  |    at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
jellyfin-1  |    at Jellyfin.Api.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
jellyfin-1  |    at Jellyfin.Api.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
jellyfin-1  |    at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
jellyfin-1  |    at Jellyfin.Api.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)
jellyfin-1  |    at Jellyfin.Api.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```
</details>

**Comment by RangerGH** [comment]
[JellyFin_log.txt](https://github.com/user-attachments/files/17594617/JellyFin_log.txt)
[JellyFin_log.txt](https://github.com/user-attachments/files/17594622/JellyFin_log.txt)
I will add my comment to this but.  I just started using JellyFin and tried to sort my list by drag/drop and got the out of bounds error which makes the playlist not very useful for lists that need to be in order.

**Comment by erictroebs** [comment]
FYI: The bug is still present in the most recent version 10.10.1.

**Comment by jc1bert** [comment]
Issue seems to be solved in 10.10.2 since Shadowghost's patch was merged.

**Comment by felix920506** [comment]
Fixed by #12934

**Comment by DickieKnee** [comment]
removing items from playlist seems to be working now... i'm on 10.10.3

https://i.postimg.cc/W1T6C1dn/JF10-3.png
Cheers.
